std/math/bits/conversion.go

```diff
@@ -56,9 +56,10 @@ type baseConversionConfig struct {
 // BaseConversionOption configures the behaviour of scalar decomposition.
 type BaseConversionOption func(opt *baseConversionConfig) error
 
-// WithNbDigits set the resulting number of digits to be used in the base conversion.
-// nbDigits must be > 0. If nbDigits is lower than the length of full
-// decomposition, then nbDigits least significant digits are returned. If the
+// WithNbDigits sets the resulting number of digits (nbDigits) to be used in the
+// base conversion. nbDigits must be > 0. If nbDigits is lower than the length of full
+// decomposition and WithUnconstrainedOutputs option is not used,
+// then a proof can not be generated. If WithNbDigits
```

Comment:
replace "proof can not be generated" by something more "constraint" oriented (ie this generates an unsatisfiable constraint)

Reply:
looks good?
```
// WithNbDigits sets the resulting number of digits (nbDigits) to be used in the base conversion.
// nbDigits must be > 0. If nbDigits is lower than the length of full decomposition and
// WithUnconstrainedOutputs option is not used, then this function generates an unsatisfiable
// constraint. If WithNbDigits option is not set, then the full decomposition is returned.
```